dashboard/utils.py

```diff
@@ -38,6 +44,7 @@ class MMTrack:
     financial_aid_min_price = None
     financial_aid_max_price = None
     financial_aid_date_documents_sent = None
+    final_grades = {}
```

Comment:
You can use a [local memory cache with Django.](https://docs.djangoproject.com/en/1.10/topics/cache/#local-memory-caching) That seems like the best of both worlds.